synapse/storage/databases/main/room.py

```diff
@@ -552,7 +552,7 @@ async def get_rooms_paginate(
             INNER JOIN room_stats_current curr USING (room_id)
             INNER JOIN rooms USING (room_id)
             %s
-            ORDER BY %s %s
+            ORDER BY %s %s, state.room_id ASC
```

Comment:
Per my other comment, should this maybe be `state.room_id %s` where the placeholder is `"ASC" if order_by_asc else "DESC"`?